### PR TITLE
plat-amlogic: Add initial support for Amlogic platforms

### PR DESCRIPTION
**Pull request by carlocaione**
This is the initial support for the Amlogic platforms.

Tested 64-bin mode on A113D (AXG) board using upstream BL31.

* xtest results (-l 15):
| 44074 subtests of which 0 failed
| 96 test cases of which 0 failed
| 0 test cases were skipped
| TEE test application done!

* Compiled with:
| make PLATFORM=amlogic

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

### CODE REVIEW

File: core/arch/arm/plat-amlogic/platform_config.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#include <mm/generic_ram_layout.h>
+
+/* Make stacks aligned to data cache line length */
+#define STACK_ALIGNMENT		64
+
+#ifdef ARM64
+#ifdef CFG_WITH_PAGER
+#error "Pager not supported for ARM64"
+#endif
+#endif /*ARM64*/
```
Comment (jforissier) [inline core/arch/arm/plat-amlogic/platform_config.h]: Is this true? Generally speaking OP-TEE supports `CFG_WITH_PAGER=y` on Arm64, I suspect those lines were copied from older code.

If you don't want pager enabled (e.g., you know it doesn't work with your platform or for whatever reason!), I'd rather have `$(call force,CFG_WITH_PAGER,n)` in `plat-amlogic/conf.mk`.

Comment (carlocaione) [reply core/arch/arm/plat-amlogic/platform_config.h]: Yeah, it was copied over and at the moment the platform doesn't use the pager so I'll set that to `n`. Thanks.

**Comment by carlocaione** [comment]
> Hi @carlocaione , thanks for the patch!
> 
> See one comment inline, and a few minor things below. Otherwise looks good tom me.

Thanks. Force pushed a new revision with all the points addressed.

File: MAINTAINERS [hunk 2]
```diff
@@ -44,6 +44,11 @@ R:	[@OP-TEE/plat-vexpress]
 S:	Maintained
 F:	core/arch/arm/plat-vexpress/
 
+AmLogic AXG (A113D)
```
Comment (carlocaione) [inline MAINTAINERS]: It looks like I forgot the alphabet. Submitting again with the proper ordering.

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

**Comment by jforissier** [comment]
@carlocaione can you please add my R-b tag? Then I can merge this. Thanks.

**Comment by jforissier** [comment]
The IBART failure is clearly unrelated (rebase needed) so let's ignore it.

**Pull request merged**